src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -299,6 +305,59 @@ public CommandBase driveToNearest() {
 
     }
 
+    public CommandBase driveUntilTipped() {
+        return cmd().onExecute(() -> {
+            move(0.0, 0.25, 0.0);
+
+        }).runsUntil(() -> Math.abs(pigeonGyro.getPitch()) > 9);
+    }
+
+    public CommandBase balance() {
+
+        PersistenceCheck balancedCheck = new PersistenceCheck(15, () -> Math.abs(pigeonGyro.getPitch()) < 7);
+        return cmd().onExecute(() -> {
+            double[] xyz = new double[3];
+            pigeonGyro.getRawGyro(xyz);
+            double pitchVelocity = xyz[0];
+            double rollVelocity = xyz[1];
+            double yawVelocity = xyz[2];
+
+            double velocityThresholdDegreesPerSec = 1.0;
+
+            double angleVelocityDegreesPerSec = map.gyro().getRotation2d().getCos() * pitchVelocity
+                    + map.gyro().getRotation2d().getSin() * rollVelocity;
+            boolean shouldStop = (pigeonGyro.getPitch() < 0.0 &&
+                    angleVelocityDegreesPerSec > velocityThresholdDegreesPerSec)
+                    || (pigeonGyro.getPitch() > 0.0 && angleVelocityDegreesPerSec < -velocityThresholdDegreesPerSec);
+
+            Logger.getInstance().recordOutput("Pitch Velocity", map.gyro().getRotation2d().getCos() * pitchVelocity);
+            Logger.getInstance().recordOutput("Roll Velocity", map.gyro().getRotation2d().getSin() * rollVelocity);
+            Logger.getInstance().recordOutput("Angle Velocity", angleVelocityDegreesPerSec);
+            Logger.getInstance().recordOutput("Pitch", pigeonGyro.getPitch());
```

Comment:
These can be removed. We'll start logging all 3 values in the Drive data layer with the other changes.